src/app/test/test.component.ts

```diff
@@ -234,11 +238,12 @@ export class TestComponent implements OnInit, AfterViewInit {
     }
   }
 
-  uploadReport(event: any): void {
-    const file: File = event.target.files[0];
+  uploadReport(event: Event): void {
+    const eventTarget = event.target as HTMLInputElement;
+    const file: FileList | null = eventTarget.files;
```

Comment:
Please rename the variable too, it was a bit confusing at first because I saw `file[0]` first.
Why is the new code better than the old code?

Reply:
> Why is the new code better than the old code?
<img src="https://media.giphy.com/media/I2QQlj7vgtT6U/giphy.gif?cid=ecf05e47x6hadsx8yipu2t15g8z9slqvod3ps6gfy401hye6&ep=v1_gifs_related&rid=giphy.gif&ct=g" height="100">

Reply:
Let me rephrase: What did you attempt to improve? I think it's less readable than what it used to be.

Reply:
I think if file is renamed to files than it is much more readable and now also typed, which was the goal of this PR.

Reply:
Yeah my bad, it was supposed to be fileList. I forgot to change it.
What I was attempting to improve is that i gave the "Event" type to the input variable of the function, but that causes an error with `event.target` saying that it's possibly null and then when I put a question mark operator after `event.target` then it tells me that `files` doesn't exist on type `EventTarget`. That's how i got to this solution

Reply:
Then this would also be a solution. It would even check if the file exists, instead of just the array. 

```suggestion
    const file: File | undefined = eventTarget.files?.[0];
```

Reply:
Yes, this seems to be working. I didn't realize the question mark operator was `?.`, I thought it was just the question mark.
Thanks for showing me that

Reply:
You're welcome!